app/validate_txs.go

```diff
@@ -83,6 +87,10 @@ func filterBlobTxs(logger log.Logger, dec sdk.TxDecoder, ctx sdk.Context, handle
 			logger.Error("decoding already checked blob transaction", "tx", tmbytes.HexBytes(coretypes.Tx(tx.Tx).Hash()), "error", err)
 			continue
 		}
+
+		// Set the tx size on the context before calling the AnteHandler
+		ctx = ctx.WithTxBytes(tx.Tx)
```

Comment:
[question] does this need to be version gated? In other words is it consensus breaking?

Reply:
https://github.com/celestiaorg/celestia-app/pull/3909#discussion_r1796032074